torch/autograd/gradcheck.py

```diff
@@ -815,6 +843,7 @@ def adjusted_atol(atol, u, v):
     #
     # We see that atol needs to be scaled by v^T M u (where M is an all-ones M x N matrix):
     # v^T M u = \sum_{i} \sum_{j} u_i * v_j = (\sum_{i} u_i)(\sum_{i} v_i)
+    u = u[0] if isinstance(u, tuple) else u
```

Comment:
Maybe a comment explaining why we ignore the second part of u? Or a todo if that needs to be done later.

Reply:
Yeah still need to think about how to handle the complex case here. I've moved the other TODO you mentioned to here.